scripts/carrying_capacity.R

```diff
@@ -1,7 +1,38 @@
+
+##' Compute the carrying capacity (equilibrium per-capita seed
+##' production) for a set of values of a trait.  Each is considered in
+##' isolation.
+##'
+##' @title Carrying Capacity
+##' @param trait_matrix A matrix of traits
+##' @param p Parameters object to use.  Importantly, the
+##' \code{strategy_default} element gets used here.
+##' @param seed_rain Initial seed rain (optional)
+##' @param parallel Use multiple processors?
+##' @author Rich FitzJohn
+##' @export
+carrying_capacity <- function(trait_matrix, p, seed_rain = 1,
+                              parallel = FALSE) {
+     f <- function(x) {
+          ## NOTE: This is not very pretty because matrix_to_list, which we
+          ## use to iterate over this, does not preserve array-ness or
+          ## names.
+          p <- expand_parameters(trait_matrix(x, traits),
+               remove_residents(p),
+               mutant = FALSE
+          )
+          p$seed_rain <- seed_rain
```

Comment:
This is now out of date - but because you've just moved to `scripts` then perhaps it's sufficient only to note that it needs revisiting? i.e. it won't break the package in the meantime.